subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -56,6 +62,30 @@ LOG_MODULE_REGISTER(net_sock_tls, CONFIG_NET_SOCKETS_LOG_LEVEL);
 #include <zephyr_mbedtls_priv.h>
 #endif
 
+#if defined(CONFIG_WOLFSSL)
+#include <user_settings.h>
+#include <wolfssl/ssl.h>
+#include <wolfssl/error-ssl.h>
+#include <wolfssl/internal.h>
```

Comment:
What is required from internal.h? It's nice to avoid including this if possible. Prefer to use API's to access members instead of direct dereference of WOLFSSL_CTX or WOLFSSL structures.

Reply:
I also am wary of using internal.h. For this instance I used it to get a decl for GetCipherNamesSize() used in tls_opt_ciphersuite_list_get(). Its more efficient as I dont have to alloc a max possible ciphernames list, only whats used. The alternative is to recode my new wolfSSL_get_cipher_list_bytes() to yield the number of ciphers if input buffer is NULL, but thats a bit inconsistent with the existing wolfSSL_get_* patterns which take a caller allocated buffer.

Reply:
I have reworked this code so as to not need internal.h anymore.